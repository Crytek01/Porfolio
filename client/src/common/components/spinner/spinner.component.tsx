import { FunctionComponent } from 'react';

import { Spin } from './spinner.styles';

interface ISpinner {
  size: number;
  isRunning?: boolean;
  className?: string;
}

export const Spinner: FunctionComponent<ISpinner> = ({
  size,
  isRunning = true,
  className,
}) => <Spin size={size} isRunning={isRunning} className={className} />;
