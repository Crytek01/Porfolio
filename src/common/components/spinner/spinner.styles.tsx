import styled from 'styled-components';

interface ISpin {
  $size: number;
  $isRunning: boolean;
}

export const Spin = styled.div<ISpin>`
  @keyframes spin {
    0% {
      transform: translate3d(-50%, 0, 0) rotate(0deg);
    }

    100% {
      transform: translate3d(-50%, 0, 0) rotate(360deg);
    }
  }

  display: block;
  animation: 1.5s linear infinite spin;
  animation-play-state: ${({ $isRunning }) =>
    $isRunning ? 'running' : 'paused'};
  border: solid 5px #cfd0d1;
  border-bottom-color: #1c87c9;
  border-radius: 50%;
  height: ${({ $size }) => $size / 10}rem;
  width: ${({ $size }) => $size / 10}rem;

  transform: translate3d(-50%, -50%, 0);
  will-change: transform;
`;
