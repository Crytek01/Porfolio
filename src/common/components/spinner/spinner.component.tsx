import { FunctionComponent } from 'react';

import { Spin } from './spinner.styles';

interface ISpinner {
  size: number;
  isRunning?: boolean;
}

export const Spinner: FunctionComponent<ISpinner> = ({
  size,
  isRunning = true,
}) => <Spin size={size} isRunning={isRunning} />;
