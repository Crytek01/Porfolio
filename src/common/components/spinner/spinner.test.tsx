import { shallowRender } from '@common/utils/test/shallowRender';
import { describe, expect, it } from 'vitest';

import { Spinner } from './spinner.component';

describe('Spinner component Test', () => {
  it('renders correctly', () => {
    const renderer = shallowRender(<Spinner size={0} />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
