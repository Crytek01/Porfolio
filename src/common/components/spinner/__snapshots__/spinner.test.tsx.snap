// Vitest Snapshot v1

exports[`Spinner component Test > renders correctly 1`] = `
<styled.div
  isRunning={true}
  size={0}
/>
`;
